File: services/interaction-service/src/main/java/com/example/intercation/controller/NotificationController.java
package com.example.intercation.controller;

import com.example.intercation.dto.request.MentionNotificationRequest;
import com.example.intercation.service.NotificationService;
import io.swagger.v3.oas.annotations.Operation;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/notifications")
@RequiredArgsConstructor
public class NotificationController {
    private final NotificationService notificationService;

    @PostMapping("/mention")
    @Operation(summary = "채널 멘션 알림")
    public ResponseEntity<Void> createMentionNotification(@RequestBody @Valid MentionNotificationRequest request) {
        notificationService.createMentionNotification(request);
        return ResponseEntity.ok().build();
    }
}

File: services/nortification-service/src/main/kotlin/com/example/nortificationservice/controller/NotificationController.kt
package com.example.nortificationservice.controller

import com.example.nortificationservice.service.NotificationService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/notifications")
class NotificationController (
    private val notificationService: NotificationService
){

    @GetMapping()
    fun home(): String {

        return "hello"
    }

}
